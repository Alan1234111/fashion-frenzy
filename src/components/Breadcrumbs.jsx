import Nav from "react-bootstrap/Nav";
import Form from "react-bootstrap/Form";
import {Link, useSearchParams} from "react-router-dom";

function Breadcrumbs() {
  const [searchParams, setSearchParams] = useSearchParams();

  function filterCategory(e) {
    setSearchParams((prevParams) => {
      if (e.target.value === "all") {
        prevParams.delete("category");
      } else {
        prevParams.set("category", e.target.value);
      }
      return prevParams;
    });
  }

  return (
    <Nav className="ms-5">
      <ul className="d-flex justify-content-around align-items-center p-0">
        <li>
          <Nav.Link as={Link} to="/" className="pe-1 ps-0 text-dark">
            Home
          </Nav.Link>
        </li>
        <li className="pe-1">&gt;</li>
        <li className="pe-1">Products</li>
        <li className="pe-1">&gt;</li>
        <li className="pe-1">
          <Form.Select onChange={(e) => filterCategory(e)} className="shadow-none pe-5 fw-bold" aria-label="Default select example">
            <option value="all">All</option>
            <option value="jewelery">Jewelery</option>
            <option value="men's clothing">Men's Clothing</option>
            <option value="women's clothing">Women's Clothing</option>
          </Form.Select>
        </li>
      </ul>
    </Nav>
  );
}

export default Breadcrumbs;
