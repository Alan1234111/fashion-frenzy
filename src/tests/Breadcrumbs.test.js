import React from "react";
import {render} from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";
import Breadcrumbs from "../components/Breadcrumbs";

test("renders Breadcrumbs correctly", () => {
  const {container} = render(
    <Router>
      <Breadcrumbs />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
