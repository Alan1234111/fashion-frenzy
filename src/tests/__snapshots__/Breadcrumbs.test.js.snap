// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Breadcrumbs correctly 1`] = `
<div>
  <div
    class="ms-md-5 nav"
  >
    <ul
      class="d-flex justify-content-around align-items-center p-0"
    >
      <li>
        <a
          class="pe-1 ps-0 text-dark nav-link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="pe-1"
      >
        &gt;
      </li>
      <li
        class="pe-1"
      >
        Products
      </li>
      <li
        class="pe-1"
      >
        &gt;
      </li>
      <li
        class="pe-1"
      >
        <select
          aria-label="Default select example"
          class="shadow-none pe-5 fw-bold form-select"
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="jewelery"
          >
            Jewelery
          </option>
          <option
            value="men's clothing"
          >
            Men's Clothing
          </option>
          <option
            value="women's clothing"
          >
            Women's Clothing
          </option>
        </select>
      </li>
    </ul>
  </div>
</div>
`;
